### ISSUE TITLE
Release jupyterhub/oauthenticator

### ISSUE THREAD

**Issue by consideRatio** [2i2c-org/infrastructure#4678]
Z2JH and TLJH are due for a major version release, including JupyterHub 5. For this, I'd like to also see us flush out an oauthenticator release.

I'm not sure if its a major or minor or patch release that's due.

## Steps

- [x] Look through the list of open PRs
- [x] Verify changelog generation with `github-activity` doesn't have PRs that can be categorized under `Other pull requests` heading.
- [x] Conclude if there are breaking changes (major version needed)
  - If so, look if something else, especially any open PRs considered breaking, should be expited to be part of the release
  - CONCLUSION: One breaking change, so v17
- [ ] Plan for a major release with others
  https://github.com/jupyterhub/oauthenticator/issues/753
- [ ] Open changelog PR (see RELEASE.md)

## Definition of done

- OAuthenticator is released according to the projects RELEASE.md

**Comment by yuvipanda** [comment]
Thanks for opening this, @consideRatio.

In order to better connect the work engineering does to the goals of the organization, I'd like you to instead:

1. Create an issue about upgrading 2i2c infrastructure to JupyterHub 5
2. We use that to *collectively* figure out what work is needed to make that happen
3. More importantly, we can use that to collectively decide what *timeline* it should happen in
4. And then we can split off that into multiple tasks, and have the *engineering team* as a whole tackle that, rather than just you as an individual.

This forms part of how I think 2i2c's overall approach should change, so we can meet the goals laid out in https://github.com/2i2c-org/meta/issues/1289. I think some of these tasks would end up as internal engineering, while some tasks would end up as product dev. For example, I'd consider getting these two PRs deployed (https://github.com/jupyterhub/kubespawner/pulls?q=is%3Apr+author%3Asunu+is%3Aclosed) to be product dev work in favor of accomplishing https://2i2c.productboard.com/roadmap/7803626-product-delivery-flow/features/26496575, and this all helps us prioritize appropriately.

**Comment by yuvipanda** [comment]
So specific action items would be:

1. Create an EPIC to say 'upgrade 2i2c infrastructure to JupyterHub' (I think you should do this)
2. Split it into subtasks that need to happen (You should also do this)
3. Prioritize these subtasks and decide on a timeline (I think right now, this is my responsibility, with input from everyone)
4. We collectively perform these tasks, scheduled through sprint planning, rather than have a single person do all of them.

**Comment by consideRatio** [comment]
I opened a release planning issue in https://github.com/jupyterhub/oauthenticator/issues/753.

This is upstream work blocking other upstream work, such as a tljh release, and that is someting i prioritize spending volunteer time working on as a jupyterhub maintainer. Due to this, I'm picking up work related to getting this (major) release out.

---

@yuvipanda I've made a personal todo-note to spend 2i2c time refining a high level 2i2c epic to upgrade to the future z2jh 4, as asked via https://github.com/2i2c-org/infrastructure/issues/4678#issuecomment-2310006686.
